source/material_model/reactive_fluid_transport.cc

```diff
@@ -475,19 +362,14 @@ namespace aspect
             }
           else if (prm.get ("Fluid-solid reaction scheme") == "tian approximation")
             {
-              AssertThrow(this->introspection().compositional_name_exists("sediment"),
-                          ExcMessage("The Tian approximation only works "
-                                     "if there is a compositional field called sediment."));
-              AssertThrow(this->introspection().compositional_name_exists("MORB"),
-                          ExcMessage("The Tian approximation only works "
-                                     "if there is a compositional field called MORB."));
-              AssertThrow(this->introspection().compositional_name_exists("gabbro"),
-                          ExcMessage("The Tian approximation only works "
-                                     "if there is a compositional field called gabbro."));
-              AssertThrow(this->introspection().compositional_name_exists("peridotite"),
-                          ExcMessage("The Tian approximation only works "
-                                     "if there is a compositional field called peridotite."));
               fluid_solid_reaction_scheme = tian_approximation;
+
+              prm.enter_subsection("Tian 2019 model");
+              {
+                tian2019_model.initialize_simulator (this->get_simulator());
+                tian2019_model.parse_parameters(prm);
+              }
+              prm.leave_subsection();
```

Comment:
So you put the parameters of the Tian model in their own subsection now. However, this means the parameter files of everyone using the reactive fluid flow model with the 3.0 version will need to be updated. Please include a changelog entry that mentions this. Additionally, you could extend our update scripts in `contrib/utilities/update_scripts/prm_files/` to automatically move this parameter. The closest operation that does something like this is probably [this function](https://github.com/geodynamics/aspect/blob/27bc5779e070c00e23f7fb951bc3a31bbe0164c8/contrib/utilities/update_scripts/prm_files/move_particles.py#L112). However, the update process looks a bit cryptic, I leave this up to you to decide if this is important enough.